quantum/rgblight.c

```diff
@@ -683,32 +683,30 @@ void rgblight_sethsv_slave(uint16_t hue, uint8_t sat, uint8_t val) {
 
 #ifndef RGBLIGHT_CUSTOM_DRIVER
 void rgblight_set(void) {
-  LED_TYPE *start_led = led + clipping_start_pos;
+  LED_TYPE led_frame[RGBLED_NUM];
```

Comment:
use a define macro to switch between the left frame buffer when RGBLIGHT_LED_MAP is enabled, and led when it is not. No reason to use this buffer if you are not remapping led indexes.

Reply:
One hypothetical case for using a separate buffer: if you want effects that are based on the previous frame. The existing logic destroys `led[]`, whereas this version does not. This memory is also allocated on the stack rather than the heap.

If there is no chance this will be useful, or a different implementation would serve the purpose better (maybe actual double buffering?), I can remove it.

Reply:
I'd remove it until a need arises for it tbh.

Reply:
Sure thing. It's not good to make things that aren't needed.